### PR TITLE
[CDAP-14620] Adds test for new navbar.

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds cypress integration tests for new navbar
- Removes unused karma tests
- Converts cypress tests from js to ts and adds webpack preprocessor plugin to help cypress understand ts.

JIRA:
 https://issues.cask.co/browse/CDAP-14620
Build:
 https://builds.cask.co/browse/CDAP-UDUT174

**Note:** 
 - Have added a new express endoint to update theme from API. This is purely used for testing purpose only for now. This helps us update the theme from API to test different theme. 
 - This is not production ready yet as we don't persist this information on node server restart

### CODE REVIEW

**Comment by elfenheart** [comment]
There seemed to be a .png file of a failed login being committed. Is that intentional?

**Comment by ajainarayanan** [comment]
@elfenheart Seems like the screenshot was because login tests were failing but still the mocha reporter reported as passed. I am confused as to why this is happening and why we didn't see this before. For now I have skipped running the tests if auth is not enabled. 

We need to rewrite the login tests. I wonder how this was tested when we merged it initially. We probably need to fetch the username and password from right place to be able to test it against a valid secure environment (like against a cluster).

File: cdap-ui/cypress/integration/login.spec.ts [hunk 1]
```diff
@@ -16,23 +16,27 @@
...
     });
   });
 
-  it('logs user in when given correct credentials', function() {
+  it('logs user in when given correct credentials', () => {
+    if (!isAuthEnabled) {
+      cy.log('Effectively skipping test as auth is not enabled');
+      return;
+    }
+    cy.log('askjakhjfbd');
```
Comment (elfenheart) [inline cdap-ui/cypress/integration/login.spec.ts]: nit: remove

**Review by elfenheart — APPROVED** [review]
minor comment, otherwise 👍 LGTM

**Comment by ajainarayanan** [comment]
Latest green build: https://builds.cask.co/browse/CDAP-UDUT174-4

**Pull request merged**